src/irmin-pack/unix/dict.ml

```diff
@@ -39,7 +39,8 @@ module Make (Fm : File_manager.S) = struct
   let append_string t v =
     let len = Int32.of_int (String.length v) in
     let buf = int32_to_bin_string len ^ v in
-    File.append_exn t buf
+    File.append_exn t buf;
+    Fm.flush_exn t.fm
```

Comment:
```suggestion
```
`File.append_exn` flushes when buffer is full, no need to do it here

But there is a flush missing in the tests, just before
https://github.com/mirage/irmin/blob/gc/test/irmin-pack/test_pack.ml#L162
as we need  flush before reloading the RO instance

Reply:
Thanks, I moved the flush into the test